activestorage/lib/active_storage/attached/many.rb

```diff
@@ -45,18 +45,27 @@ def detach
     end
 
     # Directly purges each associated attachment (i.e. destroys the blobs and
-    # attachments and deletes the files on the service).
-    def purge
+    # attachments and deletes the files on the service) unless ID(s) given.
+    def purge(*args)
       if attached?
-        attachments.each(&:purge)
-        attachments.reload
+        if args.any?
+          attachments.where(id: args).each(&:purge)
+        else
+          attachments.each(&:purge)
+          attachments.reload
+        end
       end
     end
 
-    # Purges each associated attachment through the queuing system.
-    def purge_later
+    # Purges each associated attachment through the queuing system unless ID(s) given.
+    def purge_later(*args)
       if attached?
-        attachments.each(&:purge_later)
+        if args.any?
+          attachments.where(id: args).each(&:purge)
+        else
+          attachments.each(&:purge)
+          attachments.reload
```

Comment:
I prefer `attachments.reset` than `attachments.reload` to lazy queries to the database if it is needed.

```diff
diff --git a/activestorage/lib/active_storage/attached/many.rb b/activestorage/lib/active_storage/attached/many.rb
index 3f17feae6c..b3b4c33cb5 100644
--- a/activestorage/lib/active_storage/attached/many.rb
+++ b/activestorage/lib/active_storage/attached/many.rb
@@ -57,11 +57,11 @@ def purge
     def purge_later(*args)
       if attached?
         if args.any?
-          attachments.where(id: args).each(&:purge)
+          attachments.where(id: args).each(&:purge_later)
         else
-          attachments.each(&:purge)
-          attachments.reload
+          attachments.each(&:purge_later)
         end
+        attachments.reset
       end
     end
   end
```

Reply:
`.reload` was copied over from [many.rb](https://github.com/rails/rails/blob/master/activestorage/lib/active_storage/attached/many.rb#L52) in master. `.reset` also runs green in the test suite though.